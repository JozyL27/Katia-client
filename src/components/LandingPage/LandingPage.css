#lp_main-flex {
  /* height: 100vh; */
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 30px;
}

.lp_login {
  position: absolute;
  top: .5rem;
  right: .5rem;
}

.lp_header {
  margin: 1rem 0;
}

.lp_last-header {
  margin-bottom: 2rem;
}

#Landing_Page__container {
  font-size: 1.6rem;
  /* border: .1rem solid black; */
  text-align: center;
  padding: 1rem 1rem 2rem;
  margin-top: 70px;
}

.lp_button {
  display: block;
  font-size: 1.8rem;
  border: .1rem solid black;
  /* background-color: lightgray;
  color: black; */
  padding: .7rem;
  text-decoration: none;
  text-align: center;

  width: 7.5rem;
  margin: 20px auto;
  border-radius: 4px;
  background-color: rgb(0,204,0);
  color: white;
  border-color: rgb(0,204,0);
  transition-duration: 0.4s;
}

.lp_button:hover {
  cursor: pointer;
  background-color: rgb(0,204,0);
  transform: translateY(-.25rem);
}

.lp_signup {
  width: 7.5rem;
  margin: auto;
}

.lp_lp-link {
  position: absolute;
  top: .5rem;
  left: .5rem;
  font-size: 1.8rem;
  text-decoration: none;
  color: black;
}

.lp_input-div {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  margin: .5rem;
}

.lp_input {
  width: 125px;
}


.katiaAbout {
  margin-top: 120px;
  font-size: 20px;
}