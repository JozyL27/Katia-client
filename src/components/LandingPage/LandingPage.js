import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import './LandingPage.css'
import AuthService from '../../services/auth-api-service'
import TokenService from '../../services/token-service'

class LandingPage extends Component {
  handleDemoClick = () => { // Only needed if we have a demo account setup
    const demoAccount = {
      "user_name": 'demouser',
      "password": 'password'
    }
    AuthService.postLogin(demoAccount)
      .then(res => {
        TokenService.saveAuthToken(res.authToken)
        this.props.onLoginSuccess(demoAccount.user_name)
      })
      .catch(res => {
        console.log({error: res.error})
      })
  }

  render() {
    return (
      <div id='landing-page'>
        <Link to='/login'>Login</Link>
        <h1>Project Manhattan</h1>
        <h2>Find new teammates</h2>
        <h3>who own the platforms you own</h3>
        <h3>and play the games you play</h3>
        <Link to='/signup'>Signup</Link>
      </div>
    )
  }
}

export default LandingPage
