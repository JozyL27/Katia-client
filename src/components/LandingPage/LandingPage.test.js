import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import LandingPage from './LandingPage'

describe('<LandingPage />', () => {
  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <LandingPage />
      </BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  // Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <LandingPage />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})