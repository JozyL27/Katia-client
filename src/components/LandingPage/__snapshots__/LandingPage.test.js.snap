// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> renders the UI as expected 1`] = `
<div
  id="landing-page"
>
  <a
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
  <h1>
    Project Manhattan
  </h1>
  <h2>
    Find new teammates
  </h2>
  <h3>
    who own the platforms you own
  </h3>
  <h3>
    and play the games you play
  </h3>
  <a
    href="/signup"
    onClick={[Function]}
  >
    Signup
  </a>
</div>
`;
